tests/core/test_entity.py

```diff
@@ -523,6 +523,26 @@ def test_slice(self):
         sliced = query[1:]
         assert len(sliced) == 4
 
+    def test_caching(self):
+        """Test that results are cached after query is evaluated"""
+        # Add multiple entries to the DB
+        Dog.create(id=2, name='Murdock', age=7, owner='John')
+        Dog.create(id=3, name='Jean', age=3, owner='John')
+        Dog.create(id=4, name='Bart', age=6, owner='Carrie')
+
+        # Filter by Dog attributes
+        query = Dog.query.filter(owner='John').order_by('age')
+
+        # Result cache is empty to begin with
+        assert query._result_cache is None
+
+        # Total invokes an evaluation and a query
+        assert query.total == 2
+
+        # Result cache is now populated
```

Comment:
Do we have the case where we are testing that a cache is invalidated?

Reply:
Cache invalidation (because the data underlying it has changed) is a rare case scenario. So rare that even Django suggests following the `all()1 path to refresh. Anyway, once the "request" is over, the `query` object is destroyed and a new query object with no cached data is constructed. Same applies to manipulation of the query object in any way - It will result in a new query clone object.